authorizer_lambda/src/authorizer_lambda/main.py

```diff
@@ -1,19 +1,61 @@
 from typing import Any
+
+from authorizer_lambda.cognito_adapter import fetch_owner_from_cognito
 from authorizer_lambda.config import Settings, config
 from authorizer_lambda.models import (
     AuthorizationResponse,
     OwnerResponse,
     TokenAuthorizationRequest,
     OwnerRequest,
+    PolicyDocument,
+    PolicyStatement,
+    StatementEffect,
 )
+import jwt
 
 
 def get_owner_uuid(cognito_id: str, settings: Settings) -> str:
     """Retrieve the owner_uuid from the lambda based on the cognito id and return it."""
     request = OwnerRequest(cognito_id=cognito_id)
+    cognito_object = fetch_owner_from_cognito(cognito_id)
     # get this from the lambda
     # response: OwnerResponse = ...
-    return response.owner.owner_uuid
+    return cognito_object.owner_uuid
+
+
+def verify_dev_token(jwt_token: str) -> str:
+    decoded_token = jwt.decode(jwt_token, config.dev_jwt_secret, algorithms=["HS256"])
+
+    if decoded_token.get("owner_uuid"):
+        return decoded_token["owner_uuid"]
+    else:
+        raise Exception("Invalid dev token")
+
+
+def verify_cognito_token(jwt_token: str) -> str:
+    decoded_token = jwt.decode(jwt_token, config.public_rsa_key, algorithms=["RS256"])
```

Comment:
To verify the token, you should use JWK to fetch the public key. This is documented in [the documentation that I linked to](https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-verifying-a-jwt.html#amazon-cognito-user-pools-using-tokens-step-2) from the README.md. And also I linked to the [page in the pyjwt docs](https://pyjwt.readthedocs.io/en/latest/usage.html) on how this can be implemented with pyjwt.

Reply:
I didn't know the PyJWK API, that would sure be a lot easier to implement. I just fetched from the project I mentioned where I used the assymetric key. That's awesome!